File: FlowCrypt/src/main/java/com/flowcrypt/email/api/email/model/MessageFlag.java
/*
 * © 2016-2018 FlowCrypt Limited. Limitations apply. Contact human@flowcrypt.com
 * Contributors: DenBond7
 */

package com.flowcrypt.email.api.email.model;

/**
 * The message flags. This flags will be used in the local database.
 *
 * @author DenBond7
 * Date: 20.06.2017
 * Time: 17:47
 * E-mail: DenBond7@gmail.com
 */

public class MessageFlag {
  public static final String ANSWERED = "\\ANSWERED";
  public static final String DELETED = "\\DELETED";
  public static final String DRAFT = "\\DRAFT";
  public static final String FLAGGED = "\\FLAGGED";
  public static final String RECENT = "\\RECENT";
  public static final String SEEN = "\\SEEN";
}

File: FlowCrypt/src/main/java/com/flowcrypt/email/api/email/model/MessageFlag.kt
/*
 * © 2016-present FlowCrypt a.s. Limitations apply. Contact human@flowcrypt.com
 * Contributors: DenBond7
 */

package com.flowcrypt.email.api.email.model

/**
 * The message flags. This flags will be used in the local database.
 *
 * @author Denys Bondarenko
 */
enum class MessageFlag constructor(val value: String) {
  ANSWERED("\\ANSWERED"),
  DELETED("\\DELETED"),
  DRAFT("\\DRAFT"),
  FLAGGED("\\FLAGGED"),
  RECENT("\\RECENT"),
  SEEN("\\SEEN");

  companion object {
    fun flagsToString(messageFlags: Collection<MessageFlag>): String {
      return messageFlags.joinToString(" ") { it.value }
    }
  }
}
